spec/cc/engine/processed_source_spec.rb

```diff
@@ -0,0 +1,39 @@
+require "spec_helper"
+require "cc/engine/processed_source"
+
+RSpec.describe CC::Engine::ProcessedSource, in_tmpdir: true do
+  include AnalyzerSpecHelpers
+
+  describe "#ast" do
```

Comment:
I like having tests for this class since these are public methods!

It does feel a little like we're testing the parser here, but I'm okay with it - since this code seems unlikely to change frequently - and kind of serves as a stand in integration test. Also, it would have been helpful in this case.